### PR TITLE
Bof 6 bof 7 create timer and save results to database

### PR DESCRIPTION
**Pull request by Vertonowsky**
https://best-offer-finder.atlassian.net/jira/software/projects/BOF/boards/1?selectedIssue=BOF-6

https://best-offer-finder.atlassian.net/jira/software/projects/BOF/boards/1?selectedIssue=BOF-7

### CODE REVIEW

File: src/main/java/com/example/backend/common/Collections.java [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
-import java.util.Collection;
+import java.util.*;
 
 public class Collections {
 
     public static boolean isNullOrEmpty(Collection< ? > collection) {
         return (collection == null || collection.isEmpty());
     }
 
+    public static String toString(Set<Long> lista) {
```
Comment (Rivixer) [inline src/main/java/com/example/backend/common/Collections.java]: The "lista" variable is not in English

File: src/main/java/com/example/backend/timer/OtomotoTimer.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+    /**
+     * Create dto with next pages and send POST request
+     *
+     * @param otomotoDto - dto sent to REST API
+     * @param filter currently processed filter
+     * @param page page number. There are 32 elements per page
+     * @return true if there are no more new elements and timer for certain filter should be finished
+     */
+    private boolean finishedProcessingFilterBatchly(OtomotoDto otomotoDto, Filter filter, Long page) {
```
Comment (Rivixer) [inline src/main/java/com/example/backend/timer/OtomotoTimer.java]: Misleading method name. This name means that the method should do nothing other than check whether filter processing is finished.

File: src/main/java/com/example/backend/timer/otomoto/response/ThumbnailDto.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ThumbnailDto {
+
+    @JsonProperty("x1")
+    private String x1;
+
+    @JsonProperty("x2")
+    private String x2;
```
Comment (Rivixer) [inline src/main/java/com/example/backend/timer/otomoto/response/ThumbnailDto.java]: Field comments would be useful

**Review by Rivixer — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/backend/common/Collections.java (hunk 1 above)
Comment (Vertonowsky) [reply src/main/java/com/example/backend/common/Collections.java]: Changed

File: src/main/java/com/example/backend/timer/otomoto/response/ThumbnailDto.java (hunk 3 above)
Comment (Vertonowsky) [reply src/main/java/com/example/backend/timer/otomoto/response/ThumbnailDto.java]: Added comments

File: src/main/java/com/example/backend/timer/OtomotoTimer.java (hunk 2 above)
Comment (Vertonowsky) [reply src/main/java/com/example/backend/timer/OtomotoTimer.java]: Changed method names

**Review by Rivixer — APPROVED** [review]

**Pull request merged**